irods/manager/metadata_manager.py

```diff
@@ -96,8 +96,9 @@ def meta_opts(row):
 
     def add(self, model_cls, path, meta, **opts):
         # Avoid sending request with empty argument(s)
-        if not(len(path) and len(meta.name) and len(meta.value)):
-            raise ValueError('Empty value in ' + repr(meta))
+        if type(meta.name) == 'str' and type(meta.value) == 'str':
```

Comment:
Will this ever evaluate to `True`? Check this out...
```python
>>> i = 'hi'
>>> type(i)
<class 'str'>
>>> type(i) == 'str'
False
>>> type(i) == str
True
>>> j = 2
>>> type(j)
<class 'int'>
>>> type(j) == 'str'
False
>>> type(j) == str
False
```
I think we might want this to read like this:
```suggestion
        if type(meta.name) == str and type(meta.value) == str:
```
Thoughts?

Reply:
Right , my point earlier. Also, are we giving up on throwing the ValueError?

Reply:
Maybe something like:
```
        if type(meta.name) is not str or  type(meta.value) is not str \
            or not(len(path) or not len(meta.name) or not len(meta.value)):
                raise ValueError('Empty or non-string value in parameter' + repr(meta))

```

Reply:
Looking again, I think [this  class's constructor ](https://github.com/irods/python-irodsclient/blob/04baaa8c0d3591276f335bca75648b5b5c392acb/irods/message/__init__.py#L692) does all the appropriate checks

Reply:
I'll resolve this since it's the same as the other, yes.